docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,235 @@
+The fundamental use case we want to support is out of tree controllers or tools that can manage these resources.
+
+For example, users could create clusters using tools such as Terraform, Crossplane, or Kops and run CAPI on top of intalled infrastructure.
+
+The proposal might also ease adoption of Cluster API in heavily restricted environments where the provider infrastructure for the cluster needs to be managed out of band.
+
+### Goals
+
+- Introduce support for "externally managed" cluster infrastructure consistently across Cluster API providers.
+- Any machine controller or machine infrastructure controllers must be able to keep operating like they do today.
+- Reuse existing InfraCluster CRDs in "externally managed" clusters to minimise differences between the two topologies.
+
+### Non-Goals/Future Work
+
+- Modify existing managed behaviour.
+- Automatically mark InfraCluster resources as ready (this will be up to the external management component initially).
+- Support anything other than cluster infrastructure (e.g. machines).
+
+## Proposal
+
+A new annotation `cluster.x-k8s.io/managed-by: "<name-of-system>"` is going to be defined in Cluster API core repository, which helps define and identify resources managed by external controllers. The value of the annotation will not be checked by Cluster API and is considered free form text.
+
+Infrastructure providers SHOULD respect the annotation and its contract.
+
+When this annotation is present on an InfraCluster resource, the InfraCluster controller is expected to ignore the resource and not perform any reconciliation.
+Importantly, it will not modify the resource or its status in any way.
+A predicate will be provided in the Cluster API repository to aid provider implementations in filtering resources that are externally managed.
+
+Additionally, the external management system must provide all required fields within the spec of the InfraCluster and must adhere to the CAPI provider contract and set the InfraCluster status to be ready when it is appropriate to do so.
+
+While an "externally managed" InfraCluster won't reconcile or manage the lifecycle of the cluster infrastructure, CAPI will still be able to create compute nodes within it.
+
+The machine controller must be able to operate without hard dependencies regardless of the cluster infrastructure being managed or externally managed.
+![](https://i.imgur.com/nA61XJt.png)
+
+### User Stories
+
+#### Story 1 - Alternate control plane provisioning with user managed infrastructure
+As a cluster provider I want to use CAPI in my service offering to orchestrate Kubernetes bootstrapping while letting workload cluster operators own their infrastructure lifecycle.
+
+#### Story 2 - Restricted access to cloud provider APIs
+As a cluster operator I want to use CAPI to orchestrate kubernetes bootstrapping while restricting the privileges I need to grant for my cloud provider because of organisational cloud security constraints.
```

Comment:
```suggestion
#### Story 2 - Restricted access to cloud provider APIs
As a cluster operator I want to use CAPI to orchestrate kubernetes bootstrapping while restricting the privileges I need to grant for my cloud provider because of organisational cloud security constraints.

Following from the example in Story 1, many AWS environments are tightly governed by an organisation's cloud security operations unit, and provisioning of security groups in particular is often prohibited.
```